packages/test-apps/nextjs/src/app/[[...slug]]/page.tsx

```diff
@@ -0,0 +1,31 @@
+import Experience from '@/components/Experience';
+import { getExperience } from '@/utils/getExperience';
+import { detachExperienceStyles } from '@contentful/experiences-core';
+
+type Page = {
+  params: { locale?: string; slug?: string; preview?: string };
+  searchParams: { [key: string]: string | string[] | undefined };
+};
+
+export default async function ExperiencePage({ params, searchParams }: Page) {
+  const { locale = 'en-US', slug = 'home-page' } = params || {};
+  const { isPreview, expEditorMode } = searchParams;
+  const preview = isPreview === 'true';
+  const editorMode = expEditorMode === 'true';
+  const { experience, error } = await getExperience(slug, locale, preview, editorMode);
+
+  if (error) {
+    return <div>{error.message}</div>;
+  }
+
+  const stylesheet = experience ? detachExperienceStyles(experience) : null;
+
+  //experience currently needs to be stringified manually to be passed to the component
+  const experienceJSON = experience ? JSON.stringify(experience) : null;
```

Comment:
Would it be better to return a custom error message component when it is null. I know we have something similar when the canvas errors out

Reply:
The experience being undefined could be an expected condition (IE if we are in editor mode, it will be undefined), so we don't want to error out here because of that.